### ISSUE TITLE
error: linking with `cc` failed: exit code: 1

### ISSUE THREAD

**Issue by karasjoh000** [arrayfire/arrayfire-rust#183]
I am running a example 

```rust 
#[macro_use(mem_info)]
extern crate arrayfire as af;

use std::time::Instant;
use af::*;

#[allow(unused_must_use)]
#[allow(unused_variables)]
fn main() {
    set_device(0);
    info();
    let samples = 20_000_000;
    let dims = Dim4::new(&[samples, 1, 1, 1]);

    let x = &randu::<f32>(dims);
    let y = &randu::<f32>(dims);

    let start = Instant::now();

    mem_info!("Before benchmark");

    for bench_iter in 0..100 {
        let xsqrd = &mul(x, x, false);
        let ysqrd = &mul(y, y, false);
        let xplusy = &add(xsqrd, ysqrd, false);
        let root = &sqrt(xplusy);
        let cnst = &constant(1, dims);
        let (real, imag) = sum_all(&le(root, cnst, false));
        let pi_val = real*4.0/(samples as f64);
    }

    println!("Estimated Pi Value in {:?}", start.elapsed());

    mem_info!("After benchmark");
}
```

and I am getting this error: 
```shell
error: linking with `cc` failed: exit code: 1
```

**Comment by 9prady9** [comment]
What command is giving this error? `cargo run -example`??

Were you able to run the stock examples from ArrayFire-rust crate?

**Comment by karasjoh000** [comment]
`cargo run`

**Comment by nilgoyette** [comment]
I had the same problem as @karasjoh000. I have this simple program

    use arrayfire::info_string;
    fn main() {
        println!("infoString: {}", info_string(true));
    }

but it refuses to compile because of some random link errors. After playing with it for a while, I discovered that arrayfire installation and env must be done BEFORE any `cargo` commands. It seems obvious when I write it... but you could change `AF_PATH` or `LD_LIBRARY_PATH` for some reasons after a first `cargo run` and this would always fails.

In brief, if you have link errors and you're sure that it should work, try a `cargo clean`.

**Comment by czotti** [comment]
I agree with @nilgoyette solution, at first I wasn't able to make it build after adding `AF_PATH` and update `LD_LIBRARY_PATH` and reload my terminal configuration.
I made a `cargo clean` and everything was working perfectly after.

**Comment by fmorency** [comment]
I also would like to point out that I needed to create a symlink in the ArrayFire installation folder `lib → lib64` to make Rust link properly.

**Comment by 9prady9** [comment]
@fmorency I have fixed that issue in my one of local branches, which will soon get merged into master.

**Comment by 9prady9** [comment]
Original issue has been addressed and issue mentioned by @fmorency is taken care of in #186
